src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +688,402 @@ protected function limit_list( $key, $list ) {
+				if ( empty( $args['user_id'] ) ) {
+					/**
+					 * Allow filtering whether to enable creating users using the Attendee Email.
+					 *
+					 * @since TBD
+					 *
+					 * @param boolean $create_user_from_email Whether to create the User using the Attendee Email if User ID is not set.
+					 * @param array   $args                   The arguments being set for this attendee.
+					 */
+					$create_user_from_email = (boolean) apply_filters( 'tribe_tickets_create_attendee_create_user_from_email', false, $args );
+
+					if ( $create_user_from_email ) {
+						// Create the user using the attendee email.
+						$created = wp_create_user( $args['email'], wp_generate_password( 12, false ), $args['email'] );
+
+						if ( $created && ! is_wp_error( $created ) ) {
+							/**
+							 * Allow filtering whether to send the new user information email to the new user.
+							 *
+							 * @since TBD
+							 *
+							 * @param boolean $send_new_user_info Whether to send the new user information email to the new user.
+							 * @param array   $args               The arguments being set for this attendee.
+							 */
+							$send_new_user_info = (boolean) apply_filters( 'tribe_tickets_create_attendee_create_user_from_email_send_new_user_info', false, $args );
+
+							if ( $send_new_user_info ) {
+								wp_send_new_user_notifications( $created, 'user' );
+							}
+
+							$args['user_id'] = $created;
+						}
+					}
+				}
+			}
+
+			if ( isset( $args['optout'] ) ) {
+				// Enforce a 0/1 value for the optout value.
+				$args['optout'] = (int) tribe_is_truthy( $args['optout'] );
+			}
+		}
+
+		/**
+		 * Allow filtering the arguments to set for the attendee.
+		 *
+		 * @since TBD
+		 *
+		 * @param array                         $args          List of arguments to set for the attendee.
+		 * @param array                         $attendee_data List of additional attendee data.
+		 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+		 */
+		$args = apply_filters( 'tribe_tickets_attendee_repository_set_attendee_args', $args, $attendee_data, $ticket );
+
+		// Maybe run filter if using a provider key name.
+		if ( $this->key_name ) {
+			/**
+			 * Allow filtering the arguments to set for the attendee by provider key name.
+			 *
+			 * @since TBD
+			 *
+			 * @param array                         $args          List of arguments to set for the attendee.
+			 * @param array                         $attendee_data List of additional attendee data.
+			 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+			 */
+			$args = apply_filters( 'tribe_tickets_attendee_repository_set_attendee_args_' . $this->key_name, $args, $attendee_data, $ticket );
+		}
+
+		// Remove arguments that are null.
+		$args = array_filter( $args, static function ( $value ) {
+			return ! is_null( $value );
+		} );
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 8296620b4673374e4b93e32ffdcb590b</sub>